lib/elixir/lib/module/types/infer.ex

```diff
@@ -436,17 +492,14 @@ defmodule Module.Types.Infer do
   `{boolean()} | {atom()}` will not be merged or types with variables that
   are distinct but equivalent such as `a | b when a ~ b`.
   """
-  # TODO: Translate union of all top types to dynamic()
+  def to_union([type], _context), do: type
+
   def to_union(types, context) when types != [] do
     flat_types = flatten_union(types)
 
-    if :dynamic in flat_types do
```

Comment:
Are we removing this because of the new `def subtype?(_, :dynamic, _context), do: true` clause?

Reply:
Correct!